src/pipecat/pipeline/task.py

```diff
@@ -627,13 +629,23 @@ async def _source_push_frame(self, frame: Frame, direction: FrameDirection):
 
         if isinstance(frame, EndTaskFrame):
             # Tell the task we should end nicely.
+            logger.debug(f"{self}: received end task frame {frame}")
             await self.queue_frame(EndFrame())
         elif isinstance(frame, CancelTaskFrame):
             # Tell the task we should end right away.
+            logger.debug(f"{self}: received cancel task frame {frame}")
             await self.queue_frame(CancelFrame())
         elif isinstance(frame, StopTaskFrame):
             # Tell the task we should stop nicely.
+            logger.debug(f"{self}: received stop task frame {frame}")
             await self.queue_frame(StopFrame())
+        elif isinstance(frame, InterruptionTaskFrame):
+            # Tell the task we should interrupt the pipeline. Note that we are
+            # bypassing the push queue and directly queue into the
+            # pipeline. This is in case the push task is waiting for a
```

Comment:
Not sure I quite follow this reasoning...

Is it because you want the interruption to be propagated even while we're waiting for `_wait_for_pipeline_end()`?

Reply:
if so, suggested wording fix to make that clearer:

"This is in case the push task is blocked waiting for a pipeline-ending frame to finish traversing the pipeline"

Reply:
Exactly. OK, updated!